stompserver/stompserver.go

```diff
@@ -338,52 +330,63 @@ func FWJRtrace(msg *stomp.Message) ([]string, error) {
 
 // helper function to subscribe to StompAMQ end-point
 func subscribe(endpoint string) (*stomp.Subscription, error) {
-	var err error
-	var addr string
-	var conn *stomp.Conn
-	var sub *stomp.Subscription
-
+	p := lbstomp.Config{
+		URI:         Config.StompURI,
+		Login:       Config.StompLogin,
+		Password:    Config.StompPassword,
+		Iterations:  Config.StompIterations,
+		SendTimeout: Config.StompSendTimeout,
+		RecvTimeout: Config.StompRecvTimeout,
+		Endpoint:    Config.EndpointProducer,
+		ContentType: Config.ContentType,
+		Protocol:    Config.Protocol,
+		Verbose:     Config.Verbose,
+	}
+	smgr := lbstomp.New(p)
+	log.Println(smgr.String())
 	// get connection
-	conn, addr, err = stompMgr.GetConnection()
-	if err == nil {
-		if conn != nil {
-			// subscribe to ActiveMQ topic
-			sub, err = conn.Subscribe(endpoint, stomp.AckAuto)
-			if err == nil {
-				log.Println("stomp connected to", addr)
-			} else {
-				log.Println("unable to subscribe to", endpoint, err)
-			}
-		} else {
-			log.Println("no connection to StompAMQ", addr)
-		}
-	} else {
-		log.Println("unable to connect to", addr, err)
+	conn, addr, err := smgr.GetConnection()
+	if err != nil {
+		return nil, err
+	}
+	log.Println("stomp connection", conn, addr)
+	// subscribe to ActiveMQ topic
+	sub, err := conn.Subscribe(endpoint, stomp.AckAuto)
+	if err != nil {
+		log.Println("unable to subscribe to", endpoint, err)
+		return nil, err
 	}
+	log.Println("stomp subscription", sub)
 	return sub, err
 }
 
 func server() {
 	log.Println("Stomp broker URL: ", Config.StompURI)
 	// get connection
 	sub, err := subscribe(Config.EndpointConsumer)
+	if err != nil {
+		log.Fatal(err)
+	}
+
```

Comment:
Here we are not going to retry, but exit the server ?

Reply:
yes, since we just starting the server and if connection fails something wrong already.